### ISSUE TITLE
[Bug] Invidious fails to connect to SSL-secured postgres

### ISSUE THREAD

**Issue by fignew** [iv-org/invidious#2287]
**Describe the bug**

Invidious fails to connect to a remote Postgresql server with the error `SSL_connect: error:1000012e:SSL routines:OPENSSL_internal:KEY_USAGE_BIT_INCORRECT (OpenSSL::SSL::Error)`

**Steps to Reproduce**

This is a custom invidious install on CentOS 8.4
The certficate for the pg server is signed by a local CA with trust added to the invidious machine.

**Logs**

#### Invidious error:
```
Unhandled exception:  (DB::ConnectionRefused)
  from lib/sqlite3/src/sqlite3/connection.cr:10:11 in '->'
  from /opt/crystal/share/crystal/src/primitives.cr:255:3 in 'build_resource'
  from /opt/crystal/share/crystal/src/int.cr:522:11 in 'build_database'
  from src/invidious.cr:37:1 in '__crystal_main'
  from /opt/crystal/share/crystal/src/crystal/main.cr:110:5 in 'main'
  from __libc_start_main
  from _start
  from ???
Caused by: SSL_connect: error:1000012e:SSL routines:OPENSSL_internal:KEY_USAGE_BIT_INCORRECT (OpenSSL::SSL::Error)
  from /opt/crystal/share/crystal/src/openssl/ssl/context.cr:247:11 in 'new'
  from lib/pg/src/pg/connection.cr:13:23 in '->'
  from /opt/crystal/share/crystal/src/primitives.cr:255:3 in 'build_resource'
  from /opt/crystal/share/crystal/src/int.cr:522:11 in 'build_database'
  from src/invidious.cr:37:1 in '__crystal_main'
  from /opt/crystal/share/crystal/src/crystal/main.cr:110:5 in 'main'
  from __libc_start_main
  from _start
  from ???
```

#### Postgresql-12 server error:
`could not accept SSL connection: sslv3 alert illegal parameter`

**Additional context**
Other services are using this postgres server successfully.
I have gotten Invidious working with an local-socket connection for troubleshooting.
RHEL8 has a system-wide crypto policy system in place `update-crypto-policy` but invidious still doesn't work with the LEGACY setting

**Comment by unixfox** [comment]
We recommend using postgres 10 instead.

**Comment by SamantazFox** [comment]
Can you try to connect to the DB with another program (like [pgAdmin](https://www.pgadmin.org/)) and see if this error is due to invidious itself or your setup?

**Comment by fignew** [comment]
Hi fellow foxes :fox_face:
Thanks for your comments
This configuration is connecting to my production Postgres 12 server with other services that do successfully connect.
I have pretty strict SSL configs for things (TLS1.3 only) but even with that relaxed the DB was not connectable.
I did try to connect to a test psql server on localhost but the only connection that successfully worked was over a local socket (not even hostnossl) That could be related the the Postgres 10 recommendation.
I will try with some more config iterations...

**Comment by SamantazFox** [comment]
Could you please provide a more details (major/minor/patch version) about the installed packages?
Notably: postgres, crystal, invidious (commit id) and openSSL?

If you're running on Crystal 1.0.0, could you please pull latest `master` (9b74cf3, at time of writing) ad use Crystal 1.1.1?

**Comment by SamantazFox** [comment]
bump @fignew! See my previous message ^

**Comment by technonerd33** [comment]
Was getting a similar error, trying to connect to strict ssl setup on postgres. Using ecc letsencrypt for certs and I am using the latest docker image. `OpenSSL 1.1.1l  24 Aug 2021` is the host openssl

`Caused by: SSL_connect: error:10000410:SSL routines:OPENSSL_internal:SSLV3_ALERT_HANDSHAKE_FAILURE (OpenSSL::SSL::Error)`

psql.conf
```
ssl = on
ssl_cert_file = '/var/lib/postgresql/13/data/fullchain.cer'
ssl_key_file = '/var/lib/postgresql/13/data/server.key'
ssl_ecdh_curve = 'secp521r1'
ssl_min_protocol_version = 'TLSv1.3'
ssl_dh_params_file = '/var/lib/postgresql/dh.pem'
```

I removed `ssl_ecdh_curve = 'secp521r1'`  postgres defaults to using `prime256v1` I tested both `prime256v1` `secp384r1` and they both worked.
